### PR TITLE
ref(contexts): Document context → tag transformation

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This is a method of promoting context values to tags which has since
been replaced with snuba functionality.

### CODE REVIEW

File: src/sentry/interfaces/contexts.py [hunk 1]
```diff
@@ -28,8 +28,52 @@ def contexttype(cls):
...
+    }
+    ```
+
+    and you have a tag_promotion_mapping that looks like
+
+    ```python
+    tag_promotion_mapping = {"": "{some_value}", "subkey": "{subkey}"}
+    ```
+
+    Then normilzation will result in two tags being promoted:
```
Comment (davidenwang) [inline src/sentry/interfaces/contexts.py:69]: normalization*

File: src/sentry/interfaces/contexts.py [hunk 2]
```diff
@@ -28,8 +28,52 @@ def contexttype(cls):
 
 
 class ContextType:
-    indexed_fields = None
+    tag_promotion_mapping = None
+    """
+    NOTE: This method of context promotion is considered DEPRECATED! Snuba now
+          handles promotion of context to tag or column values. It is NOT
+          recommended to use this.
```
Comment (wedamija) [inline src/sentry/interfaces/contexts.py:35]: Where does this actually happen in snuba? What I saw was promoting context values to columns, but not to tags directly

Comment (evanpurkhiser) [reply src/sentry/interfaces/contexts.py]: You're right, it is not handled at all in snuba. This isn't legacy

File: src/sentry/interfaces/contexts.py [hunk 3]
```diff
@@ -28,8 +28,48 @@ def contexttype(cls):
 
 
 class ContextType:
-    indexed_fields = None
+    tag_promotion_mapping = None
```
Comment (wedamija) [inline src/sentry/interfaces/contexts.py:31]: Should we be more explicit here and say `context_to_tag_mapping`? Promotion in this context is usually used to refer to promoting a tag/context to a column

Comment (evanpurkhiser) [reply src/sentry/interfaces/contexts.py:31]: 👍

**Review by wedamija — APPROVED** [review]

**Pull request merged**